process.go

```diff
@@ -316,44 +316,47 @@ func (p *Process) Run() {
 		Failf("%s: CoresPerTask (%d) can't be greater than maxConcurrentTasks of workflow (%d)\n", p.Name(), p.CoresPerTask, cap(p.workflow.concurrentTasks))
 	}
 
-	// By using a channel to preserve Task ordering, completed Tasks can be
-	// reaped while Tasks are still being created. Waiting for all Tasks to be
-	// spawned before processing any can cause deadlock under certain workflow
-	// architectures when there are more than BUFSIZE Tasks per process, see #81.
-	// Note: This effectively limits concurrency to BUFSIZE Tasks per process.
-	tasks := make(chan *Task, BUFSIZE)
-	go func() {
-		defer close(tasks)
-		for t := range p.createTasks() {
-			tasks <- t
-
-			// Sending FIFOs for the task
-			for oname, oip := range t.OutIPs {
-				if oip.doStream {
-					if oip.FifoFileExists() {
-						Fail("Fifo file exists, so exiting (clean up fifo files before restarting the workflow): ", oip.FifoPath())
+	// Using a slice to store unprocessed tasks allows us to receive tasks as
+	// they are produced and to maintain the correct order of IPs. This select
+	// allows us to process completed tasks as they become available. Waiting
+	// for all Tasks to be spawned before processing any can cause deadlock
+	// under certain workflow architectures when there are more than BUFSIZE
+	// Tasks per process, see #81.
+	tasksToBeProcessed := taskQueue{}
```

Comment:
I felt tasksToBeProcessed was a bit confusing. I haven't really came up with any elegant name, but I'd think tasksForWhichToSendIPs would be better at least for code clarity. Thoughts?

Reply:
That's, personally, a bit verbose for my tastes. How would `startedTasks` grab you? Combined with the other change the select then becomes `case <-startedTasks.NextTaskDone():` which I think conveys the intention that we're looking to reap tasks as they are completed. 

I can change it to `tasksForWhichToSendIPs` or whatever other name you prefer.

Reply:
`startedTasks` is great 👍 ... hints you in the right direction.